rustls/src/key.rs

```diff
@@ -97,6 +99,28 @@ impl fmt::Debug for Certificate {
     }
 }
 
+/// wrapper around internal representation of a parsed certificate. This is used in order to avoid parsing twice when specifying custom verification
+#[allow(unreachable_pub)]
+#[cfg_attr(docsrs, doc(cfg(feature = "dangerous_configuration")))]
+pub struct ParsedCertificate<'a>(webpki::EndEntityCert<'a>);
+
+impl<'a> TryFrom<&'a Certificate> for ParsedCertificate<'a> {
+    type Error = Error;
+    fn try_from(value: &'a Certificate) -> Result<ParsedCertificate<'a>, Self::Error> {
+        webpki::EndEntityCert::try_from(value.0.as_ref())
+            .map_err(crate::verify::pki_error)
+            .map(ParsedCertificate)
+    }
+}
+
+impl<'a> Deref for ParsedCertificate<'a> {
+    type Target = webpki::EndEntityCert<'a>;
```

Comment:
So one thing we've been trying to avoid is having webpki types and traits exposed through rustls' public API -- so I don't think this is the right solution here.

Reply:
Ok, i removed this impl and made the inner struct pub(crate)